### ISSUE TITLE
Enhancement: Add rule for requiring error cause in catch rethrows

### ISSUE THREAD

**Issue by pertrai1** [pertrai1/eslint-plugin-llm-core#36]
## Description

LLMs often discard the original error when re-throwing, losing valuable debugging context:

```ts
// Bad - loses original error
try {
  await fetchData();
} catch (e) {
  throw new Error('Failed to fetch data');
}

// Good - preserves cause chain  
try {
  await fetchData();
} catch (e) {
  throw new Error('Failed to fetch data', { cause: e });
}
```

## Suggested Rule Name

`preserve-error-cause`

## Implementation

Detect `throw new Error(...)` inside catch blocks that don't include `{ cause: ... }` option.

## Notes

ESLint 9.35+ has a core `preserve-caught-error` rule, but this would be a more specific variant focused on the `cause` option specifically.

This was mentioned in the README as a complementary core rule, but could be worth adding as a native rule with LLM-friendly error messages.
